src/util/tile-error-collection.js

```diff
@@ -0,0 +1,80 @@
+var Backbone = require('backbone');
+var $ = require('jquery');
+var getValue = require('./get-object-value');
+var tileErrorImage = require('../../themes/img/tile-error.svg');
+
+var TILE_ERROR_IMAGE = 'data:image/svg+xml;base64,' + window.btoa(tileErrorImage);
+
+var TileErrorCollection = Backbone.Collection.extend({
+  initialize: function () {
+    this.running = false;
+    this.queue = new Backbone.Collection();
+
+    this.on('add', this._onAdd, this);
+  },
+
+  resetErrorTiles: function () {
+    this.running = false;
+    this.queue.reset();
+
+    var models = this.filter(function (model) {
+      return !this._deletedNode(model.get('tileDomNode'));
+    }.bind(this));
+
+    this.reset(models);
+  },
+
+  getError: function () {
+    var model = this.find(function (model) {
+      return model.has('error');
+    });
+
+    return model && model.get('error');
+  },
+
+  _onAdd: function (model) {
+    var isInQueue = this.queue.findWhere({ url: model.get('url') });
+    if (!isInQueue) this.queue.add(model);
+
+    model.get('tileDomNode').src = TILE_ERROR_IMAGE;
+
+    if (!this.running) {
+      this.running = true;
+      this._getTileErrors();
+    }
+  },
+
+  _getTileErrors: function () {
+    if (!this.running) return;
+
+    var model = this.queue.find(function (model) {
+      return !model.get('checked');
+    });
+
+    if (this.getError() || !model) {
+      this.running = false;
+      return;
+    }
+
+    $.ajax({
+      url: model.get('url'),
+      success: function () {
+        model.get('tileDomNode').src = model.get('url');
+      },
+      error: function (jqXHR) {
+        var errors = getValue(jqXHR, 'responseJSON.errors_with_context', []);
+        model.set('error', errors[0]);
```

Comment:
Errors in `responseJSON.errors_with_context` are objects, in other cases ([here](https://github.com/CartoDB/cartodb.js/blob/v4/src/windshaft/map-base.js#L116-L118)) we wrap them inside [this class](https://github.com/CartoDB/cartodb.js/blob/v4/src/windshaft/error.js) so that we return a known object to users of CartoDB.js. Do you think it'd be nice to do the same / something similar here?

Reply:
That's a pretty good idea, I think we could use the same class, but I don't get why we do `this.type = error.subtype`, in the response I just have:

```
{
  "type":"limit",
  "message":"You are over platform's limits. Please contact us to know more details"
}
```

Reply:
So far, we only had "layer", "analysis" and "unknown" errors. The first two were assigned to the corresponding layer or analysis model, and we used `subtype` as the type of the error (eg: `turbocarto`). Think about an error related to a layer... we might want to display the error differently if it's an error related to the "sql" or if it has something to do with the "turbocarto"... That's why needed to know the type of error (from the subtype).

"Unknown" errors were treated as "global" errors and that's the type of the error wasn't really important... ".

[This test](https://github.com/CartoDB/cartodb.js/blob/v4/test/spec/vis/model-updater.spec.js#L642-L695) might help you understand how errors from Maps API are assigned to layer models.

Perhaps the `type` is not that important in this case. We can come back to this once you implement the change that will use this error....